File: polyglot/runtime_template_resolver/mjs/src/extractor.ts
import { logger } from './logger.js';

const log = logger.create('runtime-template-resolver', 'extractor.ts');

export function extractPlaceholders(template: string): string[] {
    log.debug('extractPlaceholders() called', { template });

    const matches = template.match(/{{([^}]+)}}/g);
    if (!matches) {
        log.debug('No placeholders found');
        return [];
    }

    const placeholders = matches.map(match => match.slice(2, -2).trim());
    log.debug('Placeholders extracted', { count: placeholders.length, placeholders });
    return placeholders;
}

File: polyglot/runtime_template_resolver/py/src/runtime_template_resolver/extractor.py
import re
from typing import List
from .logger import logger

log = logger.create("runtime_template_resolver", __file__)

def extract_placeholders(template: str) -> List[str]:
    log.debug("extract_placeholders() called", template=template)
    
    matches = re.findall(r"{{([^}]+)}}", template)
    if not matches:
        log.debug("No placeholders found")
        return []
        
    placeholders = [m.strip() for m in matches]
    log.debug("Placeholders extracted", count=len(placeholders), placeholders=placeholders)
    return placeholders
